The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/12095490634/job/33728735998) [(plain)](https://github.com/rust-lang/rust/commit/5b32f5785f791c62051ae1c82f9d8d78090e6edd/checks/33728735998/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#13 2.894 Building wheels for collected packages: reuse
#13 2.895   Building wheel for reuse (pyproject.toml): started
#13 3.136   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 3.137   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132720 sha256=026f3bb0f1aa8090b861fd0a0939cb1a782396d84c8aab7875096557d637a0f6
#13 3.137   Stored in directory: /tmp/pip-ephem-wheel-cache-5l5htb14/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#13 3.140 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#13 3.531 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#13 3.532 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 4.056 Collecting virtualenv
#13 4.056 Collecting virtualenv
#13 4.097   Downloading virtualenv-20.28.0-py3-none-any.whl (4.3 MB)
#13 4.216      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.3/4.3 MB 37.0 MB/s eta 0:00:00
#13 4.256 Collecting distlib<1,>=0.3.7
#13 4.260   Downloading distlib-0.3.9-py2.py3-none-any.whl (468 kB)
#13 4.268      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 73.9 MB/s eta 0:00:00
#13 4.301 Collecting platformdirs<5,>=3.9.1
#13 4.305   Downloading platformdirs-4.3.6-py3-none-any.whl (18 kB)
#13 4.342 Collecting filelock<4,>=3.12.2
#13 4.346   Downloading filelock-3.16.1-py3-none-any.whl (16 kB)
#13 4.426 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 4.605 Successfully installed distlib-0.3.9 filelock-3.16.1 platformdirs-4.3.6 virtualenv-20.28.0
#13 DONE 4.7s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      311232 kB
DirectMap2M:     7028736 kB
DirectMap1G:    11534336 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
WARN: currently no CI rustc builds have rustc debug assertions enabled. Please either set `rust.debug-assertions` to `false` if you want to use download CI rustc or set `rust.download-rustc` to `false`.
downloading https://static.rust-lang.org/dist/2024-11-27/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz
---
     n: usize,
     mut carry: bool,
 ) -> bool {
-
     // CHECK: mov [[TEMP:r..]], qword ptr [rsi + 8*[[IND:r..]] + 8]
     // CHECK: adc [[TEMP]], qword ptr [rdx + 8*[[IND]] + 8]
     // CHECK: mov qword ptr [rdi + 8*[[IND]] + 8], [[TEMP]]
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/tests/assembly/aarch64-naked-fn-no-bti-prolog.rs" "/checkout/tests/assembly/stack-protector/stack-protector-heuristics-effect.rs" "/checkout/tests/assembly/stack-protector/stack-protector-heuristics-effect-windows-64bit.rs" "/checkout/tests/assembly/stack-protector/stack-protector-heuristics-effect-windows-32bit.rs" "/checkout/tests/assembly/stack-protector/stack-protector-target-support.rs" "/checkout/tests/assembly/simd-intrinsic-mask-reduce.rs" "/checkout/tests/assembly/simd-intrinsic-mask-load.rs" "/checkout/tests/assembly/strict_provenance.rs" "/checkout/tests/assembly/simd/reduce-fadd-unordered.rs" "/checkout/tests/assembly/simd-intrinsic-mask-store.rs" "/checkout/tests/assembly/nvptx-linking-binary.rs" "/checkout/tests/assembly/simd-intrinsic-gather.rs" "/checkout/tests/assembly/aarch64-pointer-auth.rs" "/checkout/tests/assembly/align_offset.rs" "/checkout/tests/assembly/x86_64-function-return.rs" "/checkout/tests/assembly/nvptx-safe-naming.rs" "/checkout/tests/assembly/nvptx-kernel-abi/nvptx-kernel-args-abi-v7.rs" "/checkout/tests/assembly/s390x-backchain-toggle.rs" "/checkout/tests/assembly/slice-is_ascii.rs" "/checkout/tests/assembly/x86-return-float.rs" "/checkout/tests/assembly/x86_64-bigint-add.rs" "/checkout/tests/assembly/pie-relocation-model.rs" "/checkout/tests/assembly/manual-eq-efficient.rs" "/checkout/tests/assembly/nvptx-arch-target-cpu.rs" "/checkout/tests/assembly/nvptx-internalizing.rs" "/checkout/tests/assembly/x86_64-fortanix-unknown-sgx-lvi-inline-assembly.rs" "/checkout/tests/assembly/closure-inherit-target-feature.rs" "/checkout/tests/assembly/x86_64-typed-swap.rs" "/checkout/tests/assembly/panic-unwind-no-uwtable.rs" "/checkout/tests/assembly/pic-relocation-model.rs" "/checkout/tests/assembly/issue-83585-small-pod-struct-equality.rs" "/checkout/tests/assembly/x86_64-floating-point-clamp.rs" "/checkout/tests/assembly/niche-prefer-zero.rs" "/checkout/tests/assembly/compiletest-self-test/use-minicore-no-run.rs" "/checkout/tests/rustdoc-json/reexport/private_twice_one_inline.rs" "/checkout/tests/rustdoc-json/reexport/glob_empty_mod.rs" "/checkout/tests/rustdoc-json/reexport/macro.rs" "/checkout/tests/rustdoc-json/reexport/simple_private.rs" "/checkout/tests/rustdoc-json/reexport/auxiliary/enum_with_discriminant.rs" "/checkout/tests/rustdoc-json/reexport/auxiliary/pub-struct.rs" "/checkout/tests/rustdoc-json/reexport/auxiliary/trait_with_docs.rs" "/checkout/tests/rustdoc-json/reexport/reexport_of_hidden.rs" "/checkout/tests/rustdoc-json/reexport/same_name_different_types.rs" "/checkout/tests/rustdoc-json/reexport/glob_extern.rs" "/checkout/tests/rustdoc-json/reexport/in_root_and_mod.rs" "/checkout/tests/rustdoc-json/reexport/rename_private.rs" "/checkout/tests/rustdoc-json/reexport/simple_public.rs" "/checkout/tests/rustdoc-json/reexport/glob_collision.rs" "/checkout/tests/rustdoc-json/reexport/doc_inline_external_crate.rs" "/checkout/tests/rustdoc-json/reexport/private_two_names.rs" "/checkout/tests/rustdoc-json/reexport/synthesize_trait_with_docs.rs" "/checkout/tests/rustdoc-json/reexport/mod_not_included.rs" "/checkout/tests/rustdoc-json/reexport/pub_use_doc_hidden.rs" "/checkout/tests/rustdoc-json/reexport/rename_public.rs" "/checkout/tests/rustdoc-json/reexport/export_extern_crate_as_self.rs" "/checkout/tests/rustdoc-json/reexport/reexport_method_from_private_module.rs" "/checkout/tests/rustdoc-json/reexport/extern_crate_glob.rs" "/checkout/tests/rustdoc-json/reexport/in_root_and_mod_pub.rs" "/checkout/tests/rustdoc-json/reexport/same_type_reexported_more_than_once.rs" "/checkout/tests/rustdoc-json/reexport/glob_private.rs" "/checkout/tests/rustdoc-json/structs/plain_pub_priv.rs" "/checkout/tests/rustdoc-json/structs/field_order.rs" "/checkout/tests/rustdoc-json/structs/tuple_pub_priv.rs" "/checkout/tests/assembly/nvptx-c-abi-arg-v7.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Sat Nov 30 10:11:32 UTC 2024
  network time: Sat, 30 Nov 2024 10:11:32 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/library/core/src/num/uint_macros.rs
+++ b/library/core/src/num/uint_macros.rs
@@ -2354,7 +2354,7 @@ macro_rules! uint_impl {
             //   to generate optimal code for now, and LLVM doesn't have an equivalent intrinsic
             let (a, b) = self.overflowing_add(rhs);
             let (c, d) = a.overflowing_add(carry as $SelfT);
-            (c, b || d)
+            (c, b | d)
         }
 
         /// Calculates `self` + `rhs` with a signed `rhs`.

--- a/tests/assembly/x86_64-bigint-add.rs
+++ b/tests/assembly/x86_64-bigint-add.rs
@@ -1,0 +1,33 @@
+//@ only-x86_64
+//@ assembly-output: emit-asm
+//@ compile-flags: --crate-type=lib -O -C target-cpu=x86-64-v4
+//@ compile-flags: -C llvm-args=-x86-asm-syntax=intel
+
+#![no_std]
+#![feature(bigint_helper_methods)]
+
+// This checks that the `carrying_add` implementation successfully chains, to catch
+// issues like <https://github.com/rust-lang/rust/issues/85532#issuecomment-2495119815>
+
+// This forces the ABI to avoid the windows-vs-linux ABI differences.
+
+// CHECK-LABEL: bigint_chain_carrying_add:
+#[no_mangle]
+pub unsafe extern "sysv64" fn bigint_chain_carrying_add(
+    dest: *mut u64,
+    src1: *const u64,
+    src2: *const u64,
+    n: usize,
+    mut carry: bool,
+) -> bool {
+    // CHECK: mov [[TEMP:r..]], qword ptr [rsi + 8*[[IND:r..]] + 8]
+    // CHECK: adc [[TEMP]], qword ptr [rdx + 8*[[IND]] + 8]
+    // CHECK: mov qword ptr [rdi + 8*[[IND]] + 8], [[TEMP]]
+    // CHECK: mov [[TEMP]], qword ptr [rsi + 8*[[IND]] + 16]
+    // CHECK: adc [[TEMP]], qword ptr [rdx + 8*[[IND]] + 16]
+    // CHECK: mov qword ptr [rdi + 8*[[IND]] + 16], [[TEMP]]
+    for i in 0..n {
+        (*dest.add(i), carry) = u64::carrying_add(*src1.add(i), *src2.add(i), carry);
+    }
+    carry
+}

